app/services/indexing_service.rb

```diff
@@ -1,49 +1,15 @@
 # frozen_string_literal: true
 
 class IndexingService
-  class Error < StandardError; end
-
-  # @param [ActiveRecord::Base]
-  # @param [Schema, Hash, nil]
+  # @param [Hash]
   # @param [Boolean]
-  def self.call(resource:, schema: nil, commit: false)
-    new(resource, schema, commit).index
-  end
-
-  attr_reader :resource, :schema
-
-  def initialize(resource, schema, commit)
-    @resource = resource
-    @schema = schema || DefaultSchema.new(resource.class).schema
-    @commit = commit
+  def self.call(document, commit: false)
```

Comment:
Because this class has _two_ public methods, both of which are used, to me that makes `call` not a very descriptive method name. What if we named it something like `add_document`?